pkg/expression/expression.go

```diff
@@ -919,23 +919,25 @@ func SplitDNFItems(onExpr Expression) []Expression {
 
 // EvaluateExprWithNull sets columns in schema as null and calculate the final result of the scalar function.
 // If the Expression is a non-constant value, it means the result is unknown.
-func EvaluateExprWithNull(ctx BuildContext, schema *Schema, expr Expression) (Expression, error) {
-	if MaybeOverOptimized4PlanCache(ctx, []Expression{expr}) {
+// When to extract functional dependencies, we need to close thie skipPlanCache.
+// so it only is closed by pkg/planner/core.ExtractNotNullFromConds
+func EvaluateExprWithNull(ctx BuildContext, schema *Schema, expr Expression, skipPlanCache bool) (Expression, error) {
```

Comment:
the name is ambiguious with the literal one, how about skipPlanCacheCheck

Reply:
Done